pkg/liqonet/ipam/ipam.go

```diff
@@ -999,44 +1001,42 @@ func (liqoIPAM *IPAM) mapIPToExternalCIDR(clusterID, remoteExternalCIDR, ip stri
 	}
 
 	// Check entry existence
-	endpointMapping, exists := endpointMappings[ip]
-	if !exists {
-		// Acquire IP
+	if _, exists := endpointMappings[ip]; !exists {
+		// Create new entry
 		ipamIP, err := liqoIPAM.ipam.AcquireIP(context.TODO(), localExternalCIDR)
 		if err != nil {
 			return "", fmt.Errorf("cannot allocate a new IP for endpoint %s: %w", ip, err)
 		}
-		klog.Infof("IP %s has been acquired for endpoint %s", ipamIP.IP.String(), ip)
-		// Create new entry
-		entry := netv1alpha1.EndpointMapping{
-			IP:              ipamIP.IP.String(),
-			ClusterMappings: make(map[string]netv1alpha1.ClusterMapping),
+		endpointMappings[ip] = netv1alpha1.EndpointMapping{
+			ExternalCIDROriginalIP: ipamIP.IP.String(),
+			ClusterMappings:        make(map[string]netv1alpha1.ClusterMapping),
 		}
-		endpointMapping = entry
-		endpointMappings[ip] = entry
+		klog.Infof("%s has been acquired for endpoint %s", endpointMappings[ip].ExternalCIDROriginalIP, ip)
 	}
 
-	// Update clusterMappings
-	endpointMapping.ClusterMappings[clusterID] = netv1alpha1.ClusterMapping{}
-	endpointMappings[ip] = endpointMapping
+	if _, exists := endpointMappings[ip].ClusterMappings[clusterID]; !exists {
+		// Map IP if remote cluster has remapped local ExternalCIDR
+		newIP, err := utils.MapIPToNetwork(externalCIDR, endpointMappings[ip].ExternalCIDROriginalIP)
```

Comment:
```suggestion
		externalCIDRNattedIP, err := utils.MapIPToNetwork(externalCIDR, endpointMappings[ip].ExternalCIDROriginalIP)
```

newIP looks confusing to me.